api/src/db/models/user_models.py

```diff
@@ -244,3 +248,18 @@ class SuppressedEmail(ApiSchemaTable, TimestampMixin):
     email: Mapped[str] = mapped_column(index=True)
     reason: Mapped[str]
     last_update_time: Mapped[datetime] = mapped_column(index=True)
+
+
+class UserApiKey(ApiSchemaTable, TimestampMixin):
+    """API Key table for user authentication to the API"""
+
+    __tablename__ = "user_api_key"
+
+    token_id: Mapped[uuid.UUID] = mapped_column(UUID, primary_key=True, default=uuid.uuid4)
+    key_name: Mapped[str] = mapped_column(nullable=False)
+    key_id: Mapped[str] = mapped_column(nullable=False, comment="AWS API Gateway key identifier")
```

Comment:
FYI - the type that goes in `Mapped` can be used instead of specifying nullable/the type in the `mapped_column` function. It's fine to do just: `key_name: Mapped[str]` and it'll know that is a text column (the type string maps to) and not nullable.